src/lib/mqtt/mqtt.cpp

```diff
@@ -0,0 +1,136 @@
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Burak Karaboga
+*/
+#include <mosquitto.h>
+
+#include "logMsg/logMsg.h"
+#include "mqtt/mqtt.h"
+
+
+
+/* ****************************************************************************
+*
+* Globals
+*/
+struct mosquitto *mosq;
+const char*      mqttHostname;
+unsigned short   mqttPortNumber;
+unsigned short   mqttKeepAlivePeriod;
+
+
+
+/* ****************************************************************************
+*
+* mqttOnPublishCallback -
+* 
+*/
+void mqttOnPublishCallback(struct mosquitto *mosq, void *userdata, int mid)
+{
+  //FIXME: Do we have a usecase for the callback? To be resolved during second PR
+  LM_W(("PoC: in the publish callback"));
+}
+
+
+
+/* ****************************************************************************
+*
+* mqttInit -
+*
+*/
+void mqttInit (
+  const char*    _mqttHost,
+  unsigned short _mqttPort,
+  unsigned short _mqttKeepAlive
+)
+{
+  mqttHostname = _mqttHost;
+  mqttPortNumber = _mqttPort;
+  mqttKeepAlivePeriod = _mqttKeepAlive;
+
+  LM_I(("Initializing MQTT client."));
+  
+  mosquitto_lib_init();
+  mosq = mosquitto_new(NULL, true, NULL);
+  if (mosq == NULL)
+  {
+    LM_X(1, ("Fatal Error (Could not create Mosquitto client instance.)"));
+  }
+
+  mosquitto_publish_callback_set(mosq, mqttOnPublishCallback);
+
+  LM_I(("Connecting to MQTT Broker at %s:%d", _mqttHost, _mqttPort));
+
+  int resultCode = mosquitto_connect(mosq, _mqttHost, _mqttPort, _mqttKeepAlive);
+  if (resultCode != MOSQ_ERR_SUCCESS)
+  {
+    LM_X(2, ("Fatal Error (Could not connect to MQTT Broker (%d): %s)", resultCode, mosquitto_strerror(resultCode)));
+  }
+  
+  LM_I(("Successfully connected to MQTT Broker."));
+
+  // Starts the client loop in its own thread. The client loop processes the network traffic
+  mosquitto_loop_start(mosq);
+}
+
+
+
+/* ****************************************************************************
+*
+* sendMqttNotification -
+*
+*/
+int sendMqttNotification(const std::string& content)
+{
+  const char* msg = content.c_str();
+
+  // FIXME: unhardwire QoS, unhardwire retain - Planned for second MQTT PR
```

Comment:
Topic also has to be unhardwired. Although it would be improbable to forget about it :), maybe is a good idea to mention also in the FIXME line.

Reply:
Agreed, thanks for the warning. It is improbable but it is better to approach impossible :)

Reply:
Fixed in 4896535